.github/workflows/nix-github-actions.yml

```
@@ -0,0 +1,31 @@
+name: Nix Flake actions
+
+on:
+  pull_request:
+  push:
```

I believe that by having this, when you open PRs CI will be executed two times, one time because it's a PR and one time because a push happened.
Instead this should work:
```suggestion
on:
 pull_request:
  push:
      branches:
      - master
```
I'm not 100% sure, so you want ignore this suggestion and let's see what happens.